#! /bin/sh /usr/share/dpatch/dpatch-run
## 99_fix_epn.dpatch by Alexander Wirt <formorer@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix for segfault in epn, cherrypicked from git
## DP: cedcb4a3f9fd34e2a72988e560a884fb0d8891af

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' pkg-icinga~/base/utils.c pkg-icinga/base/utils.c
--- pkg-icinga~/base/utils.c	2012-08-29 21:50:46.000000000 +0200
+++ pkg-icinga/base/utils.c	2012-09-09 12:37:44.408419317 +0200
@@ -3601,6 +3601,7 @@
 	/* grab the first line - we should see Perl. go home if not */
 	if (fgets(buf, 80, fp) == NULL || strstr(buf, "/bin/perl") == NULL) {
 		fclose(fp);
+		return FALSE;
 	}
 
 	/* epn directives must be found in first ten lines of plugin */
